tests/unit/room.test.ts

```diff
@@ -1,14 +1,142 @@
 import { readFile } from "fs/promises";
 import WebSocket from "ws";
-// import WebSocket from "ws";
 import Browser from "../../src/Browser";
 import Client, { Host } from "../../src/Client";
 import InvalidArgumentError from "../../src/Exceptions/InvalidArgumentError";
 import Message from "../../src/Message";
 import Room from "../../src/Room";
+import User from "../../src/User";
+import type { ChatEvent } from "../../src/WebsocketEvent";
 import WebsocketEvent, { ChatEventType } from "../../src/WebsocketEvent";
 
 describe("Room", () => {
+    describe("blocking users", () => {
+        beforeEach(() => jest.resetModules());
+
+        test("should correctly block users", () => {
+            expect.assertions(2);
+            const client = new Client("stackoverflow.com");
+            const roomId = 42;
+            const room = new Room(client, roomId);
+
+            const user1 = new User(client, -1);
+            room.block(user1);
+
+            const user2 = new User(client, -2);
+            room.block(user2.id);
+
+            expect(room.isBlocked(user1)).toBe(true);
+            expect(room.isBlocked(user2)).toBe(true);
+        });
+
+        test("should unblock users after a timeout if one passed", async () => {
+            expect.assertions(2);
+
+            const client = new Client("stackoverflow.com");
+            const roomId = 42;
+            const room = new Room(client, roomId);
+
+            const user = new User(client, -1);
+
+            jest.useFakeTimers();
+
+            room.block(user, 5);
+
+            expect(room.isBlocked(user)).toBe(true);
+
+            jest.runAllTimers();
+
+            // https://github.com/facebook/jest/issues/2157
+            await Promise.resolve();
+
+            expect(room.isBlocked(user)).toBe(false);
+
+            jest.useRealTimers();
+        });
+
+        test("should correclty unblock users", () => {
```

Comment:
minor typo `correclty`

Reply:
Ah, my favourite typo! The only thing keeping me from botching letter order when I type is TypeScript... Will update shortly :) Maybe will find some more of those